selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubServiceClient.java

```diff
@@ -218,10 +218,21 @@ public String getUserPublicPrimaryEmail(String username, String password) throws
     List<Map<String, String>> properties =
         response.as(List.class, new TypeToken<List<Map<String, String>>>() {}.getType());
 
-    return properties
-        .stream()
-        .filter(map -> map.get("primary").equals("true") && map.get("visibility").equals("public"))
-        .map(map -> map.get("email"))
-        .collect(Collectors.joining());
+    if (properties.isEmpty()) {
+      throw new NoSuchElementException("The list with github emails is empty");
+    }
+
+    return filterPropertiesAndGetGithubPrimaryEmail(properties);
+  }
+
+  private String filterPropertiesAndGetGithubPrimaryEmail(List<Map<String, String>> properties) {
+    List<Map<String, String>> primaryPublicGithubEmails =
+        properties
+            .stream()
+            .filter(
+                map -> map.get("primary").equals("true") && map.get("visibility").equals("public"))
+            .collect(toList());
+
+    return primaryPublicGithubEmails.get(0).get("email");
```

Comment:
Check on `primaryPublicGithubEmails.size() > 0` to avoid **NullPointerException**.